### ISSUE TITLE
degreesDF doesn't include NA's

### ISSUE THREAD

**Issue by kaijagahm** [kaijagahm/toyModel#11]
Seems like R is using vector recycling with the way I created degreesDF. This is more disastrous than just filling in an NA with the wrong value. It's a frame shift.

If we have degrees like this (these numbers are made up and have no relation to an actual network!)

1, 4, 3, 7, 8
1, 5, 6, 9

...which then gets filled in as the following due to vector recycling:
1, 4, 3, 7, 8
1, 5, 6, 9, 1

It appears as though there was an NA in position 5 that got filled in with the value from position 1. But what if the removed individual was actually individual 2? This is a disaster waiting to happen.

This is caused by using do.call(rbind) to create a data frame right off the bat, and then pivoting to long format.

A less error-prone approach would be using rbindlist() to convert directly to a long-format data frame.

Need to also keep the individual ID's constant, though. This would be best done at the level of the graph. Hmm...

**Comment by kaijagahm** [comment]
Fixing mis-assignment of vertices was a two step process.

First, in https://github.com/kaijagahm/toyModel/commit/223c507cabc4833b05ef486e2d050e9a9caa58f5, I updated the model function to make sure nodes were getting properly labeled BEFORE dropping the deleted nodes.

Then, I took advantage of `tidygraph` to change how I was creating `degreesDF` so that nodes are associated with their identities and calculated degrees are directly associated with those identities as well.
